### PR TITLE
fix ucontext configure check

### PR DESCRIPTION
**Pull request by grooverdan**
musl ships the ucontext.h header for other purposes, but makecontext is not implemented. fix the check to detect if makecontext is implemented before enabling code using it.

on glibc x86_64 it still detects fine.
```
-- Looking for ucontext.h
-- Looking for ucontext.h - found
-- Looking for makecontext
-- Looking for makecontext - found
...
$ grep CONTEXT CMakeCache.txt
HAVE_UCONTEXT_H:INTERNAL=1
HAVE_UCONTEXT_HEADER:INTERNAL=1
```

@fabled I've taken your patch from https://git.alpinelinux.org/cgit/aports/log/main/mariadb/fix-ucontext-check.patch?showmsg=1 and I'm hoping you can release this under the BSD-new license.

Similar to other open source projects, the MariaDB Foundation needs to have shared ownership of all code that is included in the MariaDB distribution. The easiest way to achieve this is by submitting your code under the BSD-new license.

The other alternative is to sign the code contribution agreement which can be found here: https://mariadb.com/kb/en/mariadb/mca/

Please indicate in a comment below that you are contributing your new code of the whole pull request, including one or several files that are either new files or modified ones, under the BSD-new license or that you have filled out the contribution agreement and sent it.

### CODE REVIEW

**Comment by fabled** [comment]
@grooverdan I here by contribute the commit in this pull request under the BSD-new license terms.

**Comment by svoj** [comment]
Hi @grooverdan, @fabled,

Thanks for your contribution!

I can't immediately see why `ucontext.h` of musl is wrong: https://git.musl-libc.org/cgit/musl/tree/include/ucontext.h

Could you share some example where it is used for other purposes?

Thanks,
Sergey

**Comment by fabled** [comment]
@svoj The problem is that `makecontext` is not implemented even if it's defined in the header. I think the `ucontext.h' is shipped mostly for `NGREG` definition.

**Comment by svoj** [comment]
Ok, sounds reasonable. Apparently we should move away from ucontext, since it is considered non portable and even non-standard anymore.

I just have one minor wish to simplify the patch, which I'll post inline.

File: configure.cmake [hunk 1]
```diff
@@ -1107,9 +1107,12 @@ SET(CMAKE_EXTRA_INCLUDE_FILES)
...
-CHECK_INCLUDE_FILE(ucontext.h HAVE_UCONTEXT_H)
-IF(NOT HAVE_UCONTEXT_H)
-  CHECK_INCLUDE_FILE(sys/ucontext.h HAVE_UCONTEXT_H)
+CHECK_INCLUDE_FILE(ucontext.h HAVE_UCONTEXT_HEADER)
+IF(NOT HAVE_UCONTEXT_HEADER)
+  CHECK_INCLUDE_FILE(sys/ucontext.h HAVE_UCONTEXT_HEADER)
+ENDIF()
+IF(HAVE_UCONTEXT_HEADER)
+  CHECK_FUNCTION_EXISTS(makecontext HAVE_UCONTEXT_H)
 ENDIF()
```
Comment (svoj) [inline configure.cmake]: Why not to reuse `HAVE_UCONTEXT_H`, then the patch should be simpler:
```
+IF(HAVE_UCONTEXT_H)
+  CHECK_FUNCTION_EXISTS(makecontext HAVE_UCONTEXT_H)
+ENDIF()
```

**Comment by fabled** [comment]
The improved patch looks good to me!

**Comment by svoj** [comment]
Thanks!

**Pull request merged**